promql/engine.go

```diff
@@ -1435,8 +1435,11 @@ func (ev *evaluator) VectorBinop(op ItemType, lhs, rhs Vector, matching *VectorM
 		// The rhs is guaranteed to be the 'one' side. Having multiple samples
 		// with the same signature means that the matching is many-to-many.
 		if _, found := rightSigs[sig]; found {
+			matchedLabels := rs.Metric.MatchLabels(matching.On, matching.MatchingLabels...)
 			// Many-to-many matching not allowed.
-			ev.errorf("many-to-many matching not allowed: matching labels must be unique on one side")
+			ev.errorf("found multiple series for the match group %s on the right hand-side of the operation "+
+				"[or left-hand side, determine this based on matching.Card]: [include the two series here]; "+
```

Comment:
Heh, I didn't meant to literally paste the parts in `[...]`, but to replace them with computed values...

Reply:
Whoops my bad :).